File: Java/pilas/Pila.java
// Pila.java
package pilas;

import nodos.Nodo;

public class Pila<T> {

    private Nodo<T> tope;
    private int tamanio;

    public Pila() {
        tope = null;
        tamanio = 0;
    }

    public void apilar(T elemento) {
        Nodo<T> nuevo = new Nodo<>(elemento);

        if (estaVacia()) {
            tope = nuevo;
        } else {
            nuevo.setSiguiente(tope);
            tope = nuevo;
        }

        tamanio++;
    }

    public T desapilar() {
        if (estaVacia()) {
            return null;
        }

        T elemento = tope.getElemento();

        tope = tope.getSiguiente();

        tamanio--;

        return elemento;
    }

    public T consultarTope() {
        if (estaVacia()) {
            return null;
        }

        return tope.getElemento();
    }

    public int tamanio() {
        return tamanio;
    }

    public boolean estaVacia() {
        return tope == null;
    }
}

File: Python/pilas/pila.py

from nodos.nodo import Nodo

class Pila[Elemento]:
	def __init__(self) -> None:
		self.__tope: Nodo[Elemento] | None = None
		self.__tamanio = 0

	def apilar(self, elemento: Elemento) -> None:
		nuevo = Nodo(elemento)

		if self.esta_vacia():
			self.__tope = nuevo
		else:
			nuevo.set_siguiente(self.__tope)
			self.__tope = nuevo

		self.__tamanio += 1

	def desapilar(self) -> Elemento | None:
		if self.esta_vacia():
			return None

		elemento = self.__tope.get_elemento()
		self.__tope = self.__tope.get_siguiente()
		self.__tamanio -= 1

		return elemento

	def consultar_tope(self) -> Elemento | None:
		if self.esta_vacia():
			return None

		return self.__tope.get_elemento()

	def tamanio(self) -> int:
		return self.__tamanio

	def esta_vacia(self) -> bool:
		return self.__tope is None
